types/time.go

```diff
@@ -2432,12 +2432,29 @@ func microSeconds(t *MysqlTime, input string, ctx map[string]int) (string, bool)
 }
 
 func yearNumericFourDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
-	v, succ := parseDigits(input, 4)
+	return yearNumericNDigits(t, input, ctx, 4)
+}
+
+func yearNumericTwoDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
+	return yearNumericNDigits(t, input, ctx, 2)
+}
+
+func yearNumericNDigits(t *MysqlTime, input string, ctx map[string]int, n int) (string, bool) {
+	var l int
+	for l = 1; l < n+1; l++ {
```

Comment:
Variable `l` and this loop is unnecessary?
I think just parse `n` bytes is briefer and better to understand.

Reply:
The n is the max number of digits,  year like '218/10/22' is also right in MySQL, it can be 2 to 4 digits. If we just pasrse n bytes, the year must have exactly n digits. SQL like `select str_to_date('18/10/22','%Y/%m/%d') from dual` or `select str_to_date('218/10/22','%Y/%m/%d') from dual` will return null.